src/libraries/System.Console/src/System/Console.cs

```diff
@@ -331,7 +419,10 @@ public static void SetCursorPosition(int left, int top)
         {
             add
             {
-                lock (InternalSyncObject)
+                // Must be placed outside s_syncObject lock. See Out getter.
```

Comment:
Did you want an assert here as well?

Reply:
To avoid having to put an assert on every usage of `lock (Console.Out)`, I placed one general assert in `Out` getter.
If `EnsureConsoleInitialized` was moved under the lock, that assert would fire.

Reply:
Ok.